code/src/components/Input.js

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable react/button-has-type */
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import styled from 'styled-components';
+import { saveTodo } from 'reducers/todoSlice';
+
+const Input = () => {
+  const [input, setInput] = useState('')
+  const dispatch = useDispatch()
+
+  const addTask = () => {
+    console.log(`Adding ${input}`)
+
+    dispatch(saveTodo({
+      item: input,
+      done: false,
+      id: Date.now()
+    }))
+  }
+
+  return (
+    <TodoBarSection>
+      <InputBar type="text" placeholder="Enter new todo..." value={input} onChange={(event) => setInput(event.target.value)} />
+      <Button onClick={addTask}>Add!</Button>
```

Comment:
Right now you are only able to add a task by pressing the button. I think if you put this in a form with and onSubmit you can also press enter on keyboard to make it more easy to add tasks :) also if you add  setInput('') at the end of the function it will clear the text in the input area